### DISCUSSION TITLE
Can not load fonts on SSR side

### DISCUSSION THREAD

**Discussion by wizardsdevMaximK** [vercel/next.js#17676] [Help] [answered]
I'm having a problem with injecting font on SSR
At first time Page loads without particular font, which is stored localy in the public folder. Browser replaces it with some default font.
Then my local font is loaded and at that moment user feels like text is jumping.
So My question is: is it possible to load page already with needed font.
Here is my code:

**fonts file style.ts:**

```
export default `
    @font-face {
      font-display: swap;
      font-weight: normal;
      font-family: 'Segoe-UI';
      src: url('/fonts/Segoe UI/Segoe UI.ttf');
      font-style: normal;
    }
    ...
`;
```
**file _document:**

```
import fonts from 'public/static/fonts/style.ts';
...

render() {
   return (
      <Head>
       ...
         <link rel="preload" href="fonts/Segoe UI/Segoe UI.ttf" as="font" crossOrigin=""/>
       ...
     </Head>
    )
}

MyDocument.getInitialProps = async (ctx) => {
    const sheets = new ServerStyleSheet();
    const originalRenderPage = ctx.renderPage;
    try {
        ctx.renderPage = () => originalRenderPage({
            enhanceApp: App => props => sheets.collectStyles(
                <App {...props} />,
            ), 
        });
        const initialProps = await Document.getInitialProps(ctx);

        return {
            ...initialProps,
            styles: (
                <>
                    <style dangerouslySetInnerHTML={{ __html: fonts }}/>
                    {initialProps.styles}
                    {sheets.getStyleElement()}
                </>
            ),
        };
    } finally {
        sheets.seal();
    }
};
```

**Comment by jamesmosier** [comment, marked as answer]
When the page loads it still needs to go and retrieve that font via an HTTP request, that is why you're seeing a flash of unstyled text.

[This is an old article](https://www.paulirish.com/2009/fighting-the-font-face-fout/) but is still relevant in many respects today, related to `@font-face` FOUT 

There are a few follow up articles as well:
* https://www.filamentgroup.com/lab/font-loading.html
* https://www.filamentgroup.com/lab/font-events.html
